CRM/Contribute/BAO/Contribution.php

```diff
@@ -4385,37 +4385,6 @@ public static function assignProportionalLineItems($trxnParams, $trxnId, $contri
     }
   }
 
-  /**
-   * Checks if line items total amounts
-   * match the contribution total amount.
-   *
-   * @param array $params
-   *  array of order params.
-   *
-   * @throws \API_Exception
-   */
-  public static function checkLineItems(&$params) {
-    $totalAmount = $params['total_amount'] ?? NULL;
```

Comment:
This function is no longer needed because we check the total after the order is instantiated now. Everything in this function other than the check only existed to support the check being done before the order was instantiated. The issue was that because we wanted to check before we started created and the creating and the parsing was in the same loop we had to have a sub-parsing loop